docs/modules/deploy/pages/configuring-kubernetes.adoc

```diff
@@ -341,3 +341,43 @@ For the complete example, please check link:https://guides.hazelcast.org/kuberne
+
+=== Configuration
+
+Automatic cluster state management is enabled by default when Hazelcast is configured with persistence enabled and Kubernetes discovery. It can be explicitly disabled by setting Hazelcast property `hazelcast.persistence.auto.cluster.state` to `false`.
+
+Depending on the use case, setting Hazelcast property `hazelcast.persistence.auto.cluster.state.strategy` configures which cluster state will be used when members are temporarily missing from the cluster (eg pod is rescheduled or rolling restart is in progress):
+
+ - use `NO_MIGRATION` cluster state if the cluster hosts a mix of persistent and in-memory data structures or if the cluster hosts only persistent data, but you favour availability over speed of recovery. While a member is missing from the cluster, cluster state switches to `NO_MIGRATION`: this way, the first replica of partitions owned by the missing member are promoted and the data are available. When the member rejoins the cluster, persistent data can be recovered from disk and a differential sync (assuming Merkle trees are enabled, which is the case by default for persistent `IMap` and `ICache`) brings them up to speed. For in-memory data structures, a full partition sync is required.
+ - use `FROZEN` cluster state if your cluster hosts only persistent data structures and you do not mind temporarily losing availability of partitions owned by a missing member in exchange for a speedy recovery from disk. Once the member rejoins, no sync over the network is required.
+
+The default value is `NO_MIGRATION`.
+
+=== Best practices
+
+When running Hazelcast with persistence in Kubernetes, the following configuration is recommended:
+
+ - Use `/hazelcast/health/node-state` as liveness probe and `/hazelcast/health/ready` as readiness probe.
+ - In your `persistence` config, do not set a non-zero value for `rebalance-delay-seconds` property. Automatic cluster state management deals with switching to appropriate cluster state that may or may not allow partition rebalancing to occur, as is appropriate depending on the detected status of the cluster.
+ - Use `PARTIAL_RECOVERY_MOST_COMPLETE` as `cluster-data-recovery-policy` and set `auto-remove-stale-data` to `true`, to minimize the need for manual interventions. Even in edge cases in which a member performs force-start (i.e. cleans its persistent data directory and rejoins the cluster with a new identity), data can usually be recovered from backup partitions (assuming `IMap` s and `ICache` s are configured with 1 or more backups).
+ - Start Hazelcast with java option `-Dhazelcast.stale.join.prevention.duration.seconds=5`; since Kubernetes may quickly reschedule pods, the default value of `30` seconds to avoid a stale join request being processed is too high and may cause unnecessary delays in members rejoining the cluster.
```

Comment:
Start Hazelcast with the `-Dhazelcast.stale.join.prevention.duration.seconds=5` Java option. Since Kubernetes...